Cabal/doc/nix-local-build.markdown

```diff
@@ -1,1313 +0,0 @@
-    [Where are my build products](#where-are-my-build-products)?)
-
-`cabal new-exec`
-:   Workaround: if you wanted to execute GHCi, consider using
-    `cabal new-repl` instead.  Otherwise, use `-v` to find the
-    list of flags GHC is being invoked with and pass it manually.
-
-`cabal new-haddock` ([#3535](https://github.com/haskell/cabal/issues/3535))
-:   Workaround: run `cabal act-as-setup -- haddock --builddir=dist-newstyle/build/pkg-0.1`
-    (or execute the Custom setup script directly).
-
-`cabal new-install` ([#3737](https://github.com/haskell/cabal/issues/3737))
-:   Workaround: no good workaround at the moment.  (But note that
-    you no longer need to install libraries before building!)
-
-# Configuring builds with cabal.project #
-
-`cabal.project` files support a variety of options which configure the
-details of your build.  The general syntax of a `cabal.project` file
-is similar to that of a Cabal file: there are a number of fields, some
-of which live inside stanzas:
-
-~~~~
-packages: */*.cabal
-with-compiler: /opt/ghc/8.0.1/bin/ghc
-
-package cryptohash
-  optimization: False
-~~~~
-
-In general, the accepted field names coincide with the accepted
-command line flags that `cabal install` and other commands take.
-For example, `cabal new-configure --library-profiling` will
-write out a project file with `library-profiling: True`.
-
-The full configuration of a project is determined by combining
-the following sources (later entries override earlier ones):
-
-1. `~/.cabal/config` (the user-wide global configuration)
-
-2. `cabal.project` (the project configuratoin)
-
-3. `cabal.project.freeze` (the output of `cabal new-freeze`)
-
-4. `cabal.project.local` (the output of `cabal new-configure`)
-
-## Specifying the local packages ##
-
-The following top-level options specify what the local packages of a
-project are:
-
-`packages:` _package location list_ (space or comma separated, default: `./*.cabal`)
-:   Specifies the list of package locations which contain
-    the local packages to be built by this project.  Package locations
-    can take the following forms:
-
-    1. They can specify a Cabal file, or a directory containing
-       a Cabal file, e.g.,
-       `packages: Cabal cabal-install/cabal-install.cabal`
-
-    2. They can specify a glob-style wildcards, which must match one
-       or more (a) directories containing a (single) Cabal file, (b) Cabal
-       files (extension `.cabal`), or (c) ~~tarballs which
-       contain Cabal packages (extension `.tar.gz`)~~ (not implemented yet).  For example, to
-       match all Cabal files in all subdirectories, as well as the Cabal
-       projects in the parent directories `foo` and `bar`, use
-       `packages: */*.cabal ../{foo,bar}/`
-
-    3. ~~They can specify an `http`, `https` or `file` URL, representing
-       the path to a remote tarball to be downloaded and built.~~ (not
-       implemented yet)
-
-    There is no command line variant of this field; see
```

Comment:
can you add a simple snippet showing how to combine this and git submodule to do local dep fork? I know it may be out of scope, but it's really important for production usage IMHO.